bundles/org.openhab.binding.viessmann/src/main/java/org/openhab/binding/viessmann/internal/handler/ViessmannAccountHandler.java

```diff
@@ -0,0 +1,565 @@
+        super(bridge);
+        this.stateStorage = stateStorage;
+        this.httpClient = httpClient;
+        this.callbackUrl = callbackUrl;
+        this.linkRegistry = linkRegistry;
+    }
+
+    public void setInstallationGatewayId(@Nullable String newInstallation, @Nullable String newGateway) {
+    }
+
+    public void addRegisteredErrorPollingGateway(String serial, Integer pollingIntervalErrors) {
+        if (!registeredErrorPollingGateway.contains(serial)) {
+            registeredErrorPollingGateway.add(serial);
+        }
+        registeredErrorPollingInterval.put(serial, pollingIntervalErrors);
+    }
+
+    public void removeRegisteredErrorPollingGateway(String serial) {
+        registeredErrorPollingGateway.remove(serial);
+        registeredErrorPollingInterval.remove(serial);
+    }
+
+    /**
+     * get the gateways list (needed for discovery)
+     *
+     * @return a list of the all gateways
+     */
+    public List<String> getGatewaysList() {
+        // return a copy of the list, so we don't run into concurrency problems
+        return new ArrayList<>(gatewaysList);
+    }
+
+    /**
+     * get the gateways to installation map (needed for discovery)
+     *
+     * @return a map of the all gateways with installationId
+     */
+    public Map<String, Integer> getSerialToInstallationId() {
+        // return a copy of the map, so we don't run into concurrency problems
+        return new HashMap<>(serialToInstallationId);
+    }
+
+    /**
+     * get the gateways to gatewayType map (needed for discovery)
+     *
+     * @return a map of the all gateways with gatewayType
+     */
+    public Map<String, String> getSerialToGatewayType() {
+        // return a copy of the map, so we don't run into concurrency problems
+        return new HashMap<>(serialToGatewayType);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // nothing to handle here
+    }
+
+    @Override
+    public void dispose() {
+        stopViessmannBridgePolling();
+        stopViessmannErrorsPolling();
+        stopViessmannBridgeLimitReset();
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Set.of(ViessmannAccountDiscoveryService.class);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initialize Viessmann Account service");
+
+        this.config = getConfigAs(AccountConfiguration.class);
+        String storedApiCalls = this.stateStorage.get(STORED_API_CALLS);
+        if (storedApiCalls != null) {
+            apiCalls = Integer.parseInt(storedApiCalls);
+        } else {
+            apiCalls = 0;
+        }
+
+        api = new ViessmannApi(this.config.apiKey, httpClient, this.config.user, this.config.password,
+                this.config.installationId, this.config.gatewaySerial, callbackUrl);
+        api.createOAuthClientService(this);
+
+        if (api.doAuthorize()) {
```

Comment:
if the autorisation fails, the thing state is not set? 
The autorisation seems synchronous, so this should run async.